client/src/components/NewChatRoomForm/useNewChatRoom.ts

```diff
@@ -0,0 +1,44 @@
+import { useInput } from "components/Input/useInput";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { SelectType } from "./Select/Select";
+import { useSelect } from "./Select/useSelect";
+
+export const useNewChatRoom = (users: SelectType) => {
+  const [members, setMembers] = useState<string[]>([]);
+
+  const [currentMember, handleMemberChange] = useSelect("");
+  const [currentRoom, handleRoomChange] = useSelect("dm");
+  const [roomName, handleRoomNameChange] = useInput("");
+
+  useEffect(() => {
+    setMembers((members) => {
+      const isMemberThere =
+        currentMember === "" || members.includes(currentMember);
+      return isMemberThere ? members : [...members, currentMember];
+    });
+  }, [currentMember]);
+
+  const removeMember = useCallback((id: string) => {
+    setMembers((m) => m.filter((el) => el !== id));
+  }, []);
+
+  const memberNames = useMemo<string[]>(
+    () => members.map((id) => users[id]),
+    [members, users]
+  );
+
+  const rooms: SelectType = { dm: "Direct Message", channel: "Channel" };
```

Comment:
Why can't we have this in constant outside instead of defining a new object on each render